SignalMessaging/utils/ThreadUtil.m

```diff
@@ -305,7 +308,9 @@ + (ThreadDynamicInteractions *)ensureDynamicInteractionsForThread:(TSThread *)th
 
         for (TSInteraction *interaction in interactionsToDelete) {
             DDLogDebug(@"Cleaning up interaction: %@", [interaction class]);
-            [interaction removeWithTransaction:transaction];
+            [writeBlocks addObject:^(YapDatabaseReadWriteTransaction *writeTransaction) {
+                [interaction removeWithTransaction:writeTransaction];
+            }];
```

Comment:
Well there's two goals. 

One is to avoid loading extensions, and the other is to avoid opening write transactions when we can avoid it.

Another thought, since we're doing this on the write connection anyway, what about making this all async?

Reply:
> Another thought, since we're doing this on the write connection anyway, what about making this all async?

I'm not 100% sure, but I think it's necessary to do this sync otherwise the dynamic interactions will often "move around" right after we enter the conversation.

Reply:
>  the dynamic interactions will often "move around"

Yeah, good point. 

I've done as you've suggested and moved to a shared editing db connection.

I'm a little bummed that we weren't able to get rid of opening this blocking write transaction every time this method is called, but I don't have a simple solution, so I'll park that aspiration for now.

I kept the shared ui connection, since it has similar benefits, we just don't use it for ensuring dynamic interactions anymore. To be clear, I'm advocating we use shared connections almost everywhere - and work to eliminate existing uiDatabaseConnections and one-off db connections as refactoring allows or performance requirements dictate.